weave/integrations/cohere/cohere_test.py

```diff
@@ -0,0 +1,232 @@
+import asyncio
+import json
+import os
+from typing import Any, Generator
+
+import cohere
+import pytest
+
+import weave
+from weave.autopatch import autopatch, reset_autopatch
+from weave.integrations.cohere.cohere_sdk import cohere_patcher
+from weave.trace_server import trace_server_interface as tsi
+
+cohere_model = "command"  # You can change this to a specific model if needed
+
+
+def _get_call_output(call: tsi.CallSchema) -> Any:
+    call_output = call.output
+    if isinstance(call_output, str) and call_output.startswith("weave://"):
+        return weave.ref(call_output).get()
+    return call_output
+
+
+@pytest.mark.skip_clickhouse_client
+@pytest.mark.vcr(
+    filter_headers=["authorization", "x-api-key"],
+    allowed_hosts=["api.wandb.ai", "localhost"],
+)
+def test_cohere(
+    client: weave.weave_client.WeaveClient,
+) -> None:
+    api_key = os.environ.get("COHERE_API_KEY", "DUMMY_API_KEY")
+
+    cohere_client = cohere.Client(api_key=api_key)
+
+    response = cohere_client.chat(
+        model=cohere_model,
+        message="Hello, Cohere!",
+        max_tokens=1024,
+    )
+
+    exp = response.text
+    assert exp.strip() != ""
+    res = client.server.calls_query(tsi.CallsQueryReq(project_id=client._project_id()))
+    assert len(res.calls) == 1
+    call = res.calls[0]
+    assert call.exception is None and call.ended_at is not None
+    output = _get_call_output(call)
+    assert output.text == exp
```

Comment:
please also add assertions for: `started_at < ended_at`, `ended_at != None`, `exception == null`, and `op_name` are all correct. This comment applies to all tests